backend/main/src/Food/Domain/Unit.ts

```diff
@@ -0,0 +1,29 @@
+import { Err, Ok, Result, ValueObject } from "src/Shared";
+import { DomainObjectError } from "src/Shared";
+
+interface IUnitProps {
+	name: string;
+}
+
+/**
+ * This class is user class.
+ * The name of unit must be less than equal to 10.
+ */
+export class Unit extends ValueObject<IUnitProps> {
+	private constructor(props: IUnitProps) {
+		super(props);
+	}
+
```

Comment:
Maybe the constructor is no longer needed because of that it is defined in the value object as a protected modifier and inherited. But it is OK to redefine it.

Reply:
Thank you for your review! I didn't know that TypeScript doesn't need to define a constructor in a subclass. I removed it.